src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -227,10 +435,16 @@ int MPL_gavl_tree_insert(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
                     direction = SEARCH_RIGHT;
                 }
             } else {
-                /* we cannot insert the duplicate buffer */
-                gavl_tree_iptr->gavl_free_fn((void *) node_ptr->val);
+                mpl_err =
```

Comment:
I don't understand what you do here. Does the `else` section handle the case where you found a matching node in the tree? Then, what do "delete" and "merge" do and why?

Reply:
`gavl_tree_delete_node` first delete a tree node from gavl tree; then `gavl_tree_node_merge` will merge input node subnode_list with this tree node; finally, we will go to `repeat` and insert this new tree node again.

Reply:
you are right. I can add a comment here.